server/src/types/SupplierTypes.ts

```diff
@@ -0,0 +1,24 @@
+export type SupplierCreate = {
+  name: string;
+  phone: string;
+  address: string;
+  email: string;
+  password: string;
+};
+
+export type SupplierResponse = {
+  id: string;
+  name: string;
+  email: string;
+  phone: string;
+  address: string;
+};
+
+export type SupplierUpdate = {
```

Comment:
We don't need the SupplierUpdate type since it is equal to the SupplierResponse, but if you'd like to leave it for semanthics remove the password property